src/App.tsx

```diff
@@ -15,37 +16,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  SORT_BY_CHAR = 'SORT_BY_CHAR',
+  SORT_BY_LENGTH = 'SORT_BY_LENGTH',
+}
+
+type TGoods = string[];
+
+interface IGoodsProps {
+  goods: Readonly<TGoods>;
+}
+
+const Goods: FC<IGoodsProps> = ({ goods }) => (
+  <ul>
+    {goods.map(el => (
+      <li data-cy="Good" key={el}>
+        {el}
+      </li>
+    ))}
+  </ul>
+);
+
+type TSortGoods = (
+  goods: TGoods,
+  sortType: typeof SortType | string,
+  isReverse: boolean,
+) => TGoods;
+
+const sortGoods: TSortGoods = (goods, sortType, isReverse = false) => {
+  if (!sortType && isReverse) return [...goods].reverse();
+
+  const sortedGoods = [...goods].sort((a, b) => {
+    switch (sortType) {
+      case SortType.SORT_BY_CHAR:
+        return isReverse ? b.localeCompare(a) : a.localeCompare(b);
+
+      case SortType.SORT_BY_LENGTH:
+        return a.length - b.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  return isReverse && sortType === SortType.SORT_BY_LENGTH
+    ? sortedGoods.reverse()
+    : sortedGoods;
+};
+
+export const App = () => {
+  const [sortType, setSortType] = useState('');
```

Comment:
1) use ur `enum` in generic type
```suggestion
  const [sortType, setSortType] = useState<SortType>(SortType.Default);
```
2) Add default sorting
```
enum SortType {
  SORT_BY_Default = '',
  SORT_BY_CHAR = 'SORT_BY_CHAR',
  SORT_BY_LENGTH = 'SORT_BY_LENGTH',
}
```